ipy2to3.py

```diff
@@ -24,37 +53,47 @@ def replace_magic_lines(lines):
             lines[i] = "\n"
     return magic_lines
 
-def convert_ipy2to3(json_to_convert):
-    for worksheet in json_to_convert['worksheets']:
-        code_cells = filter(is_python_code_cell, worksheet['cells'])
-        for cell in code_cells:
-            magic = replace_magic_lines(cell['input'])
-            file_name = None
-            with tempfile.NamedTemporaryFile(
-                    mode = "w", delete = False) as ostream:
-                ostream.writelines(cell['input'])
-                file_name = ostream.name
-            cmd2to3 = ["2to3", "--nobackups", "--write", file_name]
-            with io.open(os.devnull, "w") as nulls:
-                subprocess.check_call(cmd2to3, stdout = nulls, stderr = nulls)
-            with io.open(file_name, mode = "r") as istream:
-                cell['input'] = istream.readlines()
-            os.remove(file_name)
-            for i, line in magic:
-                cell['input'][i] = line
+
+def convert_a_cell_ipy2to3(cell, version):
+    magic = replace_magic_lines(cell['source' if version is "old" else 'input'])
+
+    file_name = None
+
+    # save cell to file, and have the filename:
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as ostream:
+        ostream.writelines(cell['source' if version is "old" else 'input'])
+        file_name = ostream.name
+
+    # convert python2 to python3 by executing 2to3:
+    # cmd2to3 = ["C:/Anaconda3_64/Scripts/2to3.exe", "--nobackups", "--write", file_name]  # windows example
+    cmd2to3 = ["2to3", "--nobackups", "--write", file_name]
+    with io.open(os.devnull, "w") as nulls:
+        subprocess.check_call(cmd2to3, stdout=nulls, stderr=nulls)
+
+    # read converted file back to cell
+    with io.open(file_name, mode="r") as istream:
+        cell['source' if version is "old" else 'input'] = istream.readlines()
```

Comment:
These should be 'input' now because python3 is only for the new version, even if the original was 'source', right?